ast/src/gen/generic.rs

```diff
@@ -1,6 +1,34 @@
 // File automatically generated by ast/asdl_rs.py.
 
 use crate::text_size::TextRange;
+#[derive(Clone, Debug, PartialEq)]
+#[derive(is_macro::Is)]
+pub enum Ast<R = TextRange> {
+    #[is(name = "module")]
+    Mod(Mod<R>),
+    Stmt(Stmt<R>),
+    Expr(Expr<R>),
+    ExprContext(ExprContext),
+    Boolop(Boolop),
+    Operator(Operator),
+    Unaryop(Unaryop),
+    Cmpop(Cmpop),
+    Comprehension(Comprehension<R>),
+    Excepthandler(Excepthandler<R>),
+    Arguments(Arguments<R>),
+    Arg(Arg<R>),
+    Keyword(Keyword<R>),
+    Alias(Alias<R>),
+    Withitem(Withitem<R>),
+    MatchCase(MatchCase<R>),
+    Pattern(Pattern<R>),
+    TypeIgnore(TypeIgnore<R>),
+}
+impl<R> Node for Ast<R> {
+    const NAME: &'static str = "AST";
+    const FIELD_NAMES: &'static [&'static str] = &[];
+}
```

Comment:
What's the reason that the enums implement the `Node` trait. They aren't really nodes and have no fields.